engine/scheduler.go

```diff
@@ -81,7 +88,18 @@ func (sas sortableAgentStates) Len() int      { return len(sas) }
 func (sas sortableAgentStates) Swap(i, j int) { sas[i], sas[j] = sas[j], sas[i] }
 
 func (sas sortableAgentStates) Less(i, j int) bool {
-	niUnits := len(sas[i].Units)
-	njUnits := len(sas[j].Units)
-	return niUnits < njUnits || (niUnits == njUnits && sas[i].MState.ID < sas[j].MState.ID)
+	iWeight, _ := calculateAgentStatsWeight(sas[i])
```

Comment:
I see. Still we need some way to handle allocated memory vs resident (=in-use) memory. 

Off-Topic: Thats the big marketing lie from Mesosphere IMO ;) Switching from resident memory to allocated memory claims